src/config/mongoose_config_spec.erl

```diff
@@ -1218,12 +1227,25 @@ ssl_opts(pgsql, Opts) -> [{ssl_opts, Opts}];
 ssl_opts(mysql, Opts) -> Opts.
 
 process_riak_tls(KVs) ->
-    {[CACertFileOpts], SSLOpts} = proplists:split(KVs, [cacertfile]),
+    KVsWithSNI = process_tls_sni(KVs),
+    {[CACertFileOpts], SSLOpts} = proplists:split(KVsWithSNI, [cacertfile]),
     riak_ssl(SSLOpts) ++ CACertFileOpts.
 
 riak_ssl([]) -> [];
 riak_ssl(Opts) -> [{ssl_opts, Opts}].
 
+process_tls_sni(KVs) ->
+    % the SSL library expects either the atom `disabled` or a string with the SNI host
+    % as value for `server_name_indication`
+    case proplists:get_value(server_name_indication, KVs, disable) of
+        enable ->
+            SNIHost = proplists:get_value(server_name_indication_host, KVs),
+            KVs1 = proplists:delete(server_name_indication, KVs),
+            lists:keyreplace(server_name_indication_host, 1, KVs1, {server_name_indication, SNIHost});
+        disable ->
+            proplists:delete(server_name_indication_host, KVs)
```

Comment:
This is counter-intuitive, as if I don't specify this option and try to set `server_name_indication_host`, the feature will be still enabled but with the host silently removed.

Reply:
Aaah... I think you are right. In such a case, a KV list with `{server_name_indication, disable}` inserted should be returned, correct?